### PR TITLE
Bug 1162276 - Display the current timezone information when auto timezone is enabled r=eragonj

### PR DESCRIPTION
**Pull request by crh0716**
https://bugzilla.mozilla.org/show_bug.cgi?id=1162276

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ed14c11d839ee6611d8c9ddee1d07e4e41180732

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6a6979beb3e57b266a292f8256a9b9444c877796

File: apps/settings/js/panels/date_time/panel.js [hunk 1]
```diff
@@ -35,6 +38,12 @@ define(function(require) {
           this._elements.clockTime.textContent = DateTime.time;
         }.bind(this);
 
+        this._boundSetTimezoneInfo = function() {
```
Comment (EragonJ) [inline apps/settings/js/panels/date_time/panel.js]: nit : let's use arrow function here.

File: apps/settings/js/panels/date_time/panel.js [hunk 2]
```diff
@@ -35,6 +38,12 @@ define(function(require) {
...
+          // Only display the timezone info when auto time is enabled.
+          var info = DateTime.clockAutoEnabled ? DateTime.timezone : '';
+          this._elements.timezoneInfo.querySelector('span').textContent = info;
```
Comment (EragonJ) [inline apps/settings/js/panels/date_time/panel.js]: I prefer not to do a sub-query in scripts, can we distinguish them in _elements like `this._elements.timezoneInfoContainer` and `this._elements.timezoneInfo` ?

File: apps/settings/js/panels/date_time/panel.js (hunk 1 above)
Comment (EragonJ) [inline apps/settings/js/panels/date_time/panel.js]: can you also help to update the others in `onInit` function ?

File: apps/settings/js/panels/date_time/panel.js [hunk 3]
```diff
@@ -23,6 +23,9 @@ define(function(require) {
           clockTime: panel.querySelector('.clock-time'),
           timeManual: panel.querySelector('.time-manual'),
           timezone: panel.querySelector('.timezone'),
+          timezoneSelectors: [].slice.apply(
```
Comment (EragonJ) [inline apps/settings/js/panels/date_time/panel.js]: For me, `selector` means the string to match related elements, do you think we should use a better naming like `timezoneElements` ... etc ?

File: apps/settings/js/panels/date_time/panel.js [hunk 4]
```diff
@@ -56,6 +65,7 @@ define(function(require) {
       onBeforeShow: function() {
         DateTime.observe('date', this._boundSetDate);
         DateTime.observe('time', this._boundSetTime);
+        DateTime.observe('timezone', this._boundSetTimezoneInfo);
```
Comment (EragonJ) [inline apps/settings/js/panels/date_time/panel.js]: should we add `DateTime.unobserve` for timezone in `onHide` ?

File: apps/settings/js/panels/date_time/panel.js (hunk 1 above)
Comment (crh0716) [inline apps/settings/js/panels/date_time/panel.js]: sure

File: apps/settings/js/panels/date_time/panel.js (hunk 2 above)
Comment (crh0716) [inline apps/settings/js/panels/date_time/panel.js]: Added a class name `timezone-info-text`.

File: apps/settings/js/panels/date_time/panel.js (hunk 3 above)
Comment (crh0716) [inline apps/settings/js/panels/date_time/panel.js]: Renamed to `timezone-picker`.

File: apps/settings/js/panels/date_time/panel.js (hunk 4 above)
Comment (crh0716) [inline apps/settings/js/panels/date_time/panel.js]: GOOD CATCH!!

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4b63ccfef839e51756b2f2775a1893a7a187aeb6

**Pull request merged**